### DISCUSSION TITLE
Is this possible? -> !#if + :matches-path

### DISCUSSION THREAD

**Discussion by Vanerio** [uBlockOrigin/uBlock-issues#2488] [Ideas] [answered]
Hi!
I have a really long list of style filters for a webpage that matches a certain path. The problem is that when using :matches-path, the filtering speed starts to decrease a lot (there are a lot of style filters) and it causes an ugly flickering. So I think that avoiding a :matches-path every single line and doing it in a filter section with the [currently existing !#if condition](https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#if-condition), would increase filtering speed a lot.

For example, something similar to:
`!#if web.com##:matches-path(/^/coding/projects//)`
`...`
`...`
`!#endif`

Maybe then the url does not need to be checked again, and all filters inside the if condition are passed directly like:
`##.badge:style(background: rgb(167,223,146) !important;)`

Would it be possible to implement this in the future?
Is there any way to do it now that I am missing?

Thank you very much!

**Comment by u-RraaLL** [comment, marked as answer]
While the idea itself seems interesting, redesigning webpages with CSS is not uBO's main purpose.

> I have a really long list of style filters

Why don't you use an extension that was designed for this instead? Stylus, for example.

~~

Also, are you sure the page you're trying to change can't be identified by any CSS selectors on it and `:matches-path()` is your only choice?

If you're on Firefox, you can also try using HTML filtering instead if the page meets its use conditions.

**Reply by Vanerio** [reply]
> While the idea itself seems interesting, redesigning webpages with CSS is not uBO's main purpose.
> 
> > I have a really long list of style filters
> 
> Why don't you use an extension that was designed for this instead? Stylus, for example.
> 
> ~~
> 
> Also, are you use the page you're trying to change can't be identified by any CSS selectors on it and `:matches-path()` is your only choice?
> 
> If you're on Firefox, you can also try using HTML filtering instead if the page meets its use conditions.

Hi!
Thank you for answering!
It may be not its main purpose, but having all my filters integrated in just one extension I fully thrust is key for me. Unfortunately :matchs-path() is the only way for not having filter conflicts and of course it is more convenient and easy to maintain anyway.

I would like to say that this not only will increase speed with style filters, of course. I have 1600 lines in My Filters orderer by website (Each website has multiple lines, and mainly not style filters), I think this change would increase speed by a lot when filtering any website. 
For example, I have 60 filters just for youtube. Omiting a check for every line in these 1600 lines by (and maybe in every 3rd party list) !#if statements (and maybe in every 3rd party list) would increase speed a lot, just by checking only certain sections.

**Reply by u-RraaLL** [reply]
> For example, I have 60 filters just for youtube. 

Do you might sharing what type of filters are these?

Youtube is actually one of the websites that has page-specific elements or attributes on most of their pages, so I doubt all of the filters would need to be procedural.

> Omiting a check for every line in these 1600 lines by (and maybe in every 3rd party list) !#if statements (and maybe in every 3rd party list) would increase speed a lot, just by checking only certain sections.

It's a nice idea, like I said, but would likely require a lot of work. The `!#if` directive is used for compiling filters (to either include or discard filter sections), so it can't be used on a per-site basis.

Since it'd be based on a procedural filter, the delay would always be there. True, it'd likely help in cases of that many filters, but `:matches-path()` wasn't designed to be used on all of your custom filters in the first place.

gorhill generally makes changes to uBO if they'd be to the benefit of many users, based on real-world cases. Restyling websites for convenience doesn't exactly meet these requirements.

**Reply by Vanerio** [reply]
In case it is useful for you I can share my filters, but I can´t see how to do it privately in GitHub (I literally created my account just to report some uBo issues, so I am not an expert).
But as my topic was marked as answered I just think that I am the only one interested in this, so not too much to do...

**Reply by krystian3w** [reply]
Github don't have DM/PM (direct/private messages) without create small Organization and private chats in these organization.

God forbid, do not use html commentary (https://www.w3schools.com/tags/tag_comment.asp) as hiding content from newbie - cite all post or convert to issue can show up these text.

---

Stylus supports regex for URL if you must omit subdomains like "www.", where fail URL begin/start with... "...".

But Stylus fail if CSS is inside `style="..."` with `!important`, on some medium blog I saw glitch with keyframes/animation where color of element was changed and code was pasted in gist to cite:

https://sveta-slepner.medium.com/overriding-inline-styles-with-css-c6a42b9cdd16 (https://pastebin.com/raw/x3atu4mg) But that seems to have been fixed by accident for Chromium 82 or 83+ and IE10-11 (I think never work in Firefox and never checked in Edge Legacy).

**Reply by Vanerio** [reply]
I would have prefered to use uBlock only so I can have all my filters in one place (a place that I totally thrust in terms of privacy and security). Unfortunately in some webpages this is not possible due to this flickering. For now I will not install any other extension and leave my needs unfulfilled.
But I would like to thank you, krystian3w, for trying to help me!!!

**Reply by u-RraaLL** [reply]
> In case it is useful for you I can share my filters, but I can´t see how to do it privately in GitHub

You can message me on Reddit. Remove `u-` to get my nickname there.
